packages/safe-apps-sdk/src/types/messaging.ts

```diff
@@ -0,0 +1,47 @@
+import { METHODS } from '../communication/methods';
+import { SafeInfo } from './sdk';
+import { GatewayTransactionDetails, SafeBalances } from './gateway';
+
+export type Methods = keyof typeof METHODS;
```

Comment:
Not related to the PR directly but... you wouldn't need this if METHODS was an enum.

Reply:
In general types for this sdk is messy and there's a room for big improvement I believe. The main thing I don't understand how to do properly is to have a connection between method/params/responses in both public and private methods. This is the reason for cumberson constructions such as:
```
  public send = <M extends Methods, P, R>(method: M, params: P): Promise<Response<R>> => {
  ```
  
If wish/have time to look at it and see if you have an idea, would really appreciate it 🙂